src/shared/hooks/useShowGoUpButton.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useEffect, useState } from "react";
+
+export function useShowGoUpButton(): boolean {
+  const [showGoUpButton, setShowGoUpButtonValue] = useState<boolean>(false);
+
+  const setShowGoUpButton = useCallback(() => {
+    if (!showGoUpButton && window.pageYOffset > 200) {
+      setShowGoUpButtonValue(true);
+    } else if (showGoUpButton && window.pageYOffset < 100) {
+      setShowGoUpButtonValue(false);
+    }
+  }, [showGoUpButton]);
+
+  useEffect(() => {
+    document.addEventListener("scroll", setShowGoUpButton);
+    return () => {
+      document.removeEventListener("scroll", setShowGoUpButton);
+    };
+  }, [showGoUpButton]);
+
+  return showGoUpButton;
+}
```

Comment:
make addEventlistener to run only once, useRef for that, to hold the referrence to actual handler

Reply:
you can create even separate custom useEventListener